Update rtfobj to use the extension from the temp path of an embedded Package object

idiom opened the pull request:
Small change which uses the extension of an embedded object from the temp_path instead of the filename. 

In cases where the embedded object is an executable file, rtfobj will display a warning for the analyst in red. If the filename extension is altered or removed, the output formatting defaults to yellow.  The extension within the temp path can't be modified otherwise it won't be executed using the correct handler for the filetype. 

In most cases I don't think it matters as an analyst reviewing the output will review all of the artifacts, but it could be used as a method to obfuscate an embedded executable object. Example: Including many embedded objects and other artifacts that fill an analysts screen.

decalage2 commented:
Thanks a lot, that's a good catch. Wouldn't it be even better to check both extensions from filename and temp path, and flag if any of them is in the list?

idiom commented:
Yeah that's a good call. I updated the request to check the extensions from both the filename and the temp path. 

I added an additional check to compare both extensions and if they don't match print the line 'MODIFIED FILE EXTENSION'. To give analysts another hint that the file has been tampered with. If this check is too verbose or the messaging should be different let me know and I can update.

decalage2 commented:
@idiom, do you have samples where the extensions are different? Otherwise I'll just try modify a normal file.
